content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md

```diff
@@ -189,6 +190,17 @@ or `kubeadm upgrade apply`), kubeadm respects the value of `UnversionedKubeletCo
 (during `kubeadm join`, `kubeadm reset`, `kubeadm upgrade ...`), kubeadm attempts to use unversioned ConfigMap name first;
 if that does not succeed, kubeadm falls back to using the legacy (versioned) name for that ConfigMap.
 
+`UpgradeAddonsBeforeControlPlane`
+: This is introduced as a **deprecated** feature gate in v1.28. Before v1.27 (including v1.27), kubeadm will upgrade the 
+addons (including CoreDNS and kube-proxy) immediately during `kubeadm upgrade apply`, regardless of whether there are other 
+control plane instances that have not been upgraded, which may cause compatibility problems. Since v1.28, kubeadm will 
+always checks whether all the control plane instances have been upgraded before starting to upgrade the addons, this behavior 
+is applied to both `kubeadm upgrade apply` and `kubeadm upgrade node`. kubeadm determines whether a control plane instance 
+has been upgraded by checking whether the image of kube-apiserver Pod has been upgraded. You **MUST** perform control plane 
+nodes upgrade sequentially, and the addons will be upgraded after the last control plane has been upgraded. The deprecated 
+`UpgradeAddonsBeforeControlPlane` feature gate gives you a chance to keep the old upgrade behavior, but please do not rely 
```

Comment:
```suggestion
`UpgradeAddonsBeforeControlPlane` feature gate gives you a chance to keep the old upgrade behavior.
You should not need this old behavior; if you do, you should change your cluster or upgrade processes
so that you do not rely
```